### DISCUSSION TITLE
How to pass parameters at rule initialization?

### DISCUSSION THREAD

**Discussion by rmrod** [evrete/evrete#67] [Q&A]
How to pass parameters during rule initialization and store them as instance variables, both using RuleBuilder and DSL?

I would like to reuse the same rule implementation by simply changing the parameter.

```
public class SampleRule {

    private Integer scoreParam;

    @Rule
    @Where(methods = {
      @MethodPredicate(method = "testScore", args = {"$score"})
    })
    public void execute() {
    }

    public boolean testScore(Integer $score) {
        return $score > scoreParam;
    }

}
```

Thank you!

**Comment by andbi** [comment]
Hi, have a look at the `PhaseListener` annotation described in https://www.evrete.org/docs/ajr/

This way you can initialize session or knowledge instance variables which can be both static or non-static.
